kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java

```diff
@@ -279,23 +280,21 @@ private void ackMessages() {
         ackIdsBatch.addAll(deliveredAckIds);
         deliveredAckIds.clear();
       }
-      ListenableFuture<Empty> response = subscriber.ackMessages(requestBuilder.build());
-      Futures.addCallback(
-          response,
-          new FutureCallback<Empty>() {
-            @Override
-            public void onSuccess(Empty result) {
-              ackIdsInFlight.removeAll(ackIdsBatch);
-              log.trace("Successfully acked a set of messages. {}", ackIdsBatch.size());
-            }
-
-            @Override
-            public void onFailure(Throwable t) {
-              deliveredAckIds.addAll(ackIdsBatch);
-              ackIdsInFlight.removeAll(ackIdsBatch);
-              log.error("An exception occurred acking messages: " + t);
-            }
-          });
+      final ApiFuture<Empty> response = subscriber.ackMessages(requestBuilder.build());
+      response.addListener(new Runnable() {
+        @Override
+        public void run() {
+          try {
+            response.get();
+            log.trace("Successfully acked a set of messages. {}", ackIdsBatch.size());
+          } catch (Exception e) {
+            deliveredAckIds.addAll(ackIdsBatch);
```

Comment:
I find `deliveredAckIds` to be a slightly confusing name in this context since they are pending ack ids to be acked to the server. Perhaps we can re-name to `receivedAckIds`?

Reply:
That's from a previous PR that was already merged in and so it isn't actually part of this change. Looks like the view might have been set to look at the changes for a single commit?